controllers/mdutil/util.go

```diff
@@ -548,6 +548,17 @@ func NewMSNewReplicas(deployment *clusterv1.MachineDeployment, allMSs []*cluster
 		// Do not exceed the number of desired replicas.
 		scaleUpCount = integer.Int32Min(scaleUpCount, *(deployment.Spec.Replicas)-*(newMS.Spec.Replicas))
 		return *(newMS.Spec.Replicas) + scaleUpCount, nil
+	case clusterv1.OnDeleteMachineDeploymentStrategyType:
```

Comment:
we should update the header of this function since this strategy is not using maxSurge

Reply:
addressed